src/manifest.rs

```diff
@@ -144,5 +156,119 @@ pub struct DependencyManifest {
     /// Version requirement in the buffrs format, currently only supports pinning
     pub version: VersionReq,
     /// Artifactory repository to pull dependency from
-    pub repository: String,
+    pub repository: Repository,
+    /// Artifactory registry to pull from
+    pub registry: RegistryUri,
+}
+
+#[derive(Debug, Clone, Hash, Serialize, Deserialize, PartialEq, Eq)]
+pub struct RegistryUri(Url);
+
+impl std::ops::Deref for RegistryUri {
+    type Target = Url;
+
+    fn deref(&self) -> &Self::Target {
+        &self.0
+    }
+}
+
+impl DerefMut for RegistryUri {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.0
+    }
+}
+
+impl Display for RegistryUri {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "{}", self.0)
+    }
+}
+
+impl FromStr for RegistryUri {
+    type Err = eyre::Error;
+
+    fn from_str(value: &str) -> Result<Self, Self::Err> {
+        let slf = Self(Url::from_str(value)?);
+        slf.sanity_check_url()?;
```

Comment:
This is dangerous as it assumes *every* registry ends with `/artifactory` – could you please move it back to the artifactory code + make it only happening if the host is `*.jfrog.io`?